.github/workflows/ci.yml

```diff
@@ -42,370 +42,70 @@ jobs:
       - name: Checkout Pex
         uses: actions/checkout@v3
         with:
           # We need branches and tags for some ITs.
           fetch-depth: 0
-          path: repo
-      - name: Setup Python ${{ join(matrix.python-version, '.') }}
-        uses: pantsbuild/actions/pyenv@1356087bfd1be04670c2ffde4ba94a9c6898ecd7
-        with:
-          python-version: "${{ join(matrix.python-version, '.') }}"
-      - name: Expose Pythons
-        uses: pantsbuild/actions/expose-pythons@e63d2d0e3c339bdffbe5e51e7c39550e3bc527bb
-        with:
-          skip: "${{ steps.calculate-pythons-to-expose.outputs.skip }}"
-      - name: Cache Pyenv Interpreters
-        uses: actions/cache@v3
-        with:
-          path: ${{ env._PEX_TEST_DEV_ROOT }}/pyenv
-          key: ${{ matrix.os }}-${{ runner.arch }}-pex-test-dev-root-pyenv-v1
-      - name: Cache Devpi Server
-        uses: actions/cache@v3
-        with:
-          path: ${{ env._PEX_TEST_DEV_ROOT }}/devpi
-          # We're using a key suffix / restore-keys prefix trick here to get an updatable cache.
-          # See: https://github.com/actions/cache/blob/main/tips-and-workarounds.md#update-a-cache
-          key: ${{ matrix.os }}-${{ runner.arch }}-${{ env._TOX_ENV }}-pex-test-dev-root-devpi-v1-${{ github.run_id }}
-          restore-keys: ${{ matrix.os }}-${{ runner.arch }}-${{ env._TOX_ENV }}-pex-test-dev-root-devpi-v1
+      # Some ITs need this for VCS URLs of the form git+ssh://git@github.com/...
       - name: Setup SSH Agent
         uses: webfactory/ssh-agent@v0.5.4
         env:
           SSH_PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
         if: env.SSH_PRIVATE_KEY != ''
         with:
           ssh-private-key: ${{ env.SSH_PRIVATE_KEY }}
-      - name: Run Integration Tests
+      - name: Run Unit Tests
         run: |
-          pip install -U tox
-          tox -c repo/tox.ini -e ${{ env._TOX_ENV }} -- ${{ env._PEX_TEST_POS_ARGS }}
-  pypy-integration-tests:
-    name: (PyPy ${{ join(matrix.pypy-version, '.') }}) Pip ${{ matrix.pip-version }} TOXENV=pypy${{ join(matrix.pypy-version, '') }}-integration
+          BASE_MODE=pull CACHE_MODE=pull ./dtox.sh -e ${{ matrix.tox-env }} -- ${{ env._PEX_TEST_POS_ARGS }}
+  mac-tests:
+    name: tox -e ${{ matrix.tox-env }}
     needs: org-check
-    runs-on: ${{ matrix.os }}
+    runs-on: macos-12
     strategy:
       matrix:
         include:
-          # N.B.: We need 20.04 for PyPy 2.7 tool cache support.
-          - os: ubuntu-20.04
-            pypy-version: [ 2, 7 ]
-            pip-version: 20
-          - os: ubuntu-22.04
-            pypy-version: [ 3, 9 ]
-            pip-version: 20
-          - os: ubuntu-22.04
-            pypy-version: [ 3, 9 ]
-            pip-version: 22_3_1
-          - os: ubuntu-22.04
-            pypy-version: [ 3, 9 ]
-            pip-version: 23_1_2
-    env:
-      _TOX_ENV: pypy${{ join(matrix.pypy-version, '') }}-pip${{ matrix.pip-version }}-integration
+          - python-version: [ 3, 11 ]
```

Comment:
I dropped CPython 2.7 tests for Mac to keep things simple and because its almost a certainty Mac (~I need the latest hot thing, and they do not support their old things) + Ancient Python is just not a thing.